cpp/src/decisiontree/decisiontree.cuh

```diff
@@ -185,18 +186,20 @@ tl::Tree<T, T> build_treelite_tree(const DT::TreeMetaDataNode<T, L>& rf_tree,
       ++cur_front;
 
       if (!q_node.IsLeaf()) {
-        tl_tree.AddChilds(tl_node_id);
+        const int cleft  = tl_tree.AllocNode();
+        const int cright = tl_tree.AllocNode();
+        tl_tree.SetChildren(tl_node_id, cleft, cright);
```

Comment:
Really like this change! Future-proofs us for re-arranging the nodes in new layouts efficiently.